Plug-Ins/NovaTools/System Editor/SystemWindowController.swift

```diff
@@ -0,0 +1,335 @@
+        }
+    }
+
+    @objc func resourceNameChanged(_ notification: Notification) {
+        guard let resource = notification.object as? Resource else {
+            return
+        }
+        if resource.typeCode == "sÿst" || resource.typeCode == "spöb" {
+            stellarViews[resource.id]?.updateFrame()
+            if let i = self.row(for: resource) {
+                stellarTable.reloadData(forRowIndexes: [i], columnIndexes: [1])
+            }
+            systemView.needsDisplay = true
+        }
+    }
+
+    @objc func resourceDataChanged(_ notification: Notification) {
+        guard !systemView.isSavingStellar, let resource = notification.object as? Resource else {
+            return
+        }
+        if resource.typeCode == "spöb" {
+            self.read(stellar: resource)
+            self.updateStellarList()
+            self.syncSelectionFromView()
+            systemView.needsDisplay = true
+        } else if resource.typeCode == "sÿst" {
+            self.reload()
+            self.syncSelectionFromView()
+            systemView.needsDisplay = true
+        } else if ["spïn", "rlëD", "PICT"].contains(resource.typeCode) {
+            self.syncSelectionFromView()
+            systemView.needsDisplay = true
+        }
+    }
+}
+
+extension SystemWindowController: NSTableViewDataSource, NSTableViewDelegate {
+    func numberOfRows(in tableView: NSTableView) -> Int {
+        stellarList.count + 1
+    }
+
+    func tableView(_ tableView: NSTableView, isGroupRow row: Int) -> Bool {
+        row == 0
+    }
+
+    func tableView(_ tableView: NSTableView, shouldSelectRow row: Int) -> Bool {
+        row != 0
+    }
+
+    func tableView(_ tableView: NSTableView, viewFor tableColumn: NSTableColumn?, row: Int) -> NSView? {
+        let identifier = tableColumn?.identifier ?? NSUserInterfaceItemIdentifier("HeaderCell")
+        let view = tableView.makeView(withIdentifier: identifier, owner: self) as! NSTableCellView
+        if let tableColumn {
+            if let stellar = stellarList[row - 1] {
+                switch tableColumn.identifier.rawValue {
+                case "index":
+                    view.textField?.stringValue = "#\(row)"
+                case "id":
+                    view.textField?.stringValue = "\(stellar.id)"
+                case "name":
+                    view.textField?.stringValue = stellar.name
+                default:
+                    break
+                }
+            } else {
+                switch tableColumn.identifier.rawValue {
+                case "index":
+                    view.textField?.stringValue = "#\(row)"
+                case "id":
+                    view.textField?.stringValue = "-1"
+                case "name":
+                    view.textField?.stringValue = "(unused)"
+                default:
+                    break
+                }
+            }
+        } else {
+            view.textField?.stringValue = resource.name
```

Comment:
If you add the resource name to the window title, you probably won't need it here as well. So you could put something like "Nav Defaults", or even drop the group row entirely.

Just for some background here, when I first started upgrading the galaxy with editing features, there was one global galaxy editor and I imagined it could edit systems across multiple open files. The system list would be divided into groups by file. I eventually decided this was a bad idea and reworked it to make the editor belong to the file. It will still _show_ systems from other open files, but these are greyed out and can't be selected or modified. The filename header in the table ended up remaining, even though it's not really needed.

Reply:
Updated to read "Nav Defaults" (I think that's clearer than removing entirely).